### PR TITLE
e2e; Phase out VLAN-NSM

### PR DESCRIPTION
**Pull request by zolug**
## Description

Replace vlan-nsm with network-attachment-definition in Kind operator e2e test environment to provide cluster breakout functionality for FEs.

The environment:
Creates additional docker network to be used to for external connectivity between Frontends and external-host containers.

The network-attachment-definition utilizes bridge CNI to create the external interface in the FEs. While allocation of IP addresses is achieved via Whereabouts IPAM CNI. As part of the environment a VLAN aware Linux bridge is setup on each Kind worker to be used by the bridge CNI.
    
Note:
In order for BIRD routing suite (in Frontends) to be able to get hold of a Router ID, the IPAM configuration is expected to assign a feasible IPv4 address even in case of IPv6 only deployments. Otherwise, frontend containers get stuck in CrashLoopBackoff.

## Issue link
N/A

## Checklist

- Purpose
    - [ ] Bug fix
    - [ ] New functionality
    - [ ] Documentation
    - [ ] Refactoring
    - [x] CI
- Test
    - [ ] Unit test
    - [x] E2E Test
    - [x] Tested manually
- Introduce a breaking change
    - [ ] Yes (description required)
    - [x] No

### CODE REVIEW

File: docs/demo/scripts/kind/Makefile [hunk 1]
```diff
@@ -122,9 +127,49 @@ kind-create: temp-dir kind kind-delete ## Create the Kind cluster
...
+.PHONY: whereabouts
+whereabouts: temp-dir ## Install whereabouts IPAM CNI plugin
+	git clone https://github.com/k8snetworkplumbingwg/whereabouts $(TEMP_DIR)/whereabouts && \
+	kubectl apply \
+	-f $(TEMP_DIR)/whereabouts/doc/crds/daemonset-install.yaml \
+	-f $(TEMP_DIR)/whereabouts/doc/crds/whereabouts.cni.cncf.io_ippools.yaml \
+	-f $(TEMP_DIR)/whereabouts/doc/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml && \
+	kubectl rollout status daemonset whereabouts -n kube-system --timeout 300s || \
+	{ echo >&2 "Error: Failed to install Whereabouts"; rm -fr $(TEMP_DIR)/whereabouts; exit 1; } ; \
+	rm -fr $(TEMP_DIR)/whereabouts
```
Comment (LionelJouin) [inline docs/demo/scripts/kind/Makefile:139]: Direct links could be used:
https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/daemonset-install.yaml
https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_ippools.yaml
https://raw.githubusercontent.com/k8snetworkplumbingwg/whereabouts/master/doc/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml

File: docs/demo/scripts/kind/Makefile [hunk 2]
```diff
@@ -122,9 +127,49 @@ kind-create: temp-dir kind kind-delete ## Create the Kind cluster
...
+
+.PHONY: multus
+multus: ## Install multus
+	kubectl apply -f https://raw.githubusercontent.com/k8snetworkplumbingwg/multus-cni/master/deployments/multus-daemonset.yml && \
+	kubectl rollout status daemonset kube-multus-ds -n kube-system --timeout 300s || { echo >&2 "Error: Failed to install Multus"; exit 1; }
+
+.PHONY: cni-plugins
+cni-plugins: temp-dir ## Install CNI plugins (add recovery option by failing cni install container if wget fails)
+	wget -qO $(TEMP_DIR)/cni-install.yml.j2 https://raw.githubusercontent.com/k8snetworkplumbingwg/multus-cni/master/e2e/templates/cni-install.yml.j2 && \
```
Comment (LionelJouin) [inline docs/demo/scripts/kind/Makefile:148]: why not applying (kubectl apply -f https://...) it directly with the URL? instead of downloading the file

**Review by LionelJouin — COMMENTED** [review]
I didn't really understand why we need to have the bridge network setup script, and why not using the vlan cni plugin directly instead of the bridge one.

**Comment by zolug** [comment]
> I didn't really understand why we need to have the bridge network setup script, and why not using the vlan cni plugin directly instead of the bridge one.

Bridge allows more flexibility when it comes to collocation of Frontends or re-use of VLAN ID. Thus the environment is less restrictive.

**Review by LionelJouin — APPROVED** [review]

**Pull request merged**